### DISCUSSION TITLE
Offer for contribution

### DISCUSSION THREAD

**Discussion by R3dByt3** [hydrostack/hydro#50] [General]
Hi,

I just wanted to inform you that your project looks really impressive and promising. For the duration of a year I had to work with Blazor and I couldn't help but to think that Blazor is just fine for internal quick and dirty projects.

Your approach seems far more professional to me in contrast to what Microsoft made.

Long story short: I would really love to use this library at work and in order to ease my mind about its future, I wanna offer you my help if needed.

Short about me:
Mainly C# dev with 7 years of experience and Bachelor of Science degree in applied information technology.

Feel free to contact me if you need another maintainer/contributor.

**Comment by kjeske** [comment]
Hi! Thank you for kind words about Hydro! ❤️ Thank you also for offering your help, that will be definitely needed :) I need to organise the planned features in a form of Github Project, so it will be easier to pick up things to work on. Also I'm releasing soon the Hydro Tutorial with live examples, so there will be many needs for adding more examples. I will get back to you when those things are ready! For now feel free to create regular issues when you encounter bugs or when you have suggestions for improvements, and also PRs for them.

**Reply by R3dByt3** [reply]
Glad to hear!

Regarding the live examples I might have an idea or proposal.

What do you think about porting the react library "Shadcn" or parts of it. These are free to copy, have a nice look / utility and might also cover the aspects of a live demo + some kind of functionality tests of your project.

I feel like if these were included in hydro, the library could become a fully fledged UI framework.

**Reply by kjeske** [reply]
The tutorial I'm working on will contain examples of components like text fields, buttons, drop-downs, autocomplete, tables, and others, will be based on daisyUI, and will show the source code easy to copy, so it will be a bit similar to what Shadcn offers. I agree with you that such idea is great.

In the beginning it will not cover all the features that Shadcn has, but of course with time there will be more and more. Soon I will upload the tutorial source code to a new repository and start hosting it on fly.io in at least 3 different regions, so the latency is not a problem in the main parts of the world.

**Comment by AdamJachocki** [comment]
Hi, the project looks very promising to me. Although I don't have much spare time, I would like to get deeper into this and help as much as I can. I am senior .NET developer with 20 years of experience in professional programming (various technologies). Also have my own blog (in polish: https://masterbranch.pl). So if you need someone like me, just let me know :)

**Reply by kjeske** [reply]
Hello! Thank you for reaching out! :) Whenever you want you can take a look at the list of issues, and if you have any ideas, or you want to code, that's awesome. The easiest way to learn how Hydro works, and to improve it, is to attach the source code locally to your project and check the process step by step in the debugger.

The most important thing in Hydro is the communication between the components via events. I did my best to make it easy and powerful, but there are some places for improvements. Once you start any project you be able to achieve most of the use cases, but you will also notice that there is stuff to do in this area. So the thing I'm trying to fix right now is #73, which makes it possible to handle the events even simpler.

Another important thing is #116, which will make it possible to create external UI libraries.
